pandas/tests/base/test_unique.py

```diff
@@ -105,3 +105,24 @@ def test_nunique_null(null_obj, index_or_series_obj):
         num_unique_values = len(obj.unique())
         assert obj.nunique() == max(0, num_unique_values - 1)
         assert obj.nunique(dropna=False) == max(0, num_unique_values)
+
+
+@pytest.mark.parametrize(
+    "idx_or_series_w_bad_unicode", [pd.Index(["\ud83d"]), pd.Series(["\ud83d"])]
+)
+def test_unique_bad_unicode(idx_or_series_w_bad_unicode):
+    # regression test for #34550
+    obj = idx_or_series_w_bad_unicode
+    obj = np.repeat(obj, range(1, len(obj) + 1))
```

Comment:
Are we trying to add duplicates here? If so probably makes sense to add them up top.